htdocs/mrp/mo_production.php

```diff
@@ -149,19 +149,14 @@
 		$result = $object->setStatut($object::STATUS_INPROGRESS, 0, '', 'MRP_REOPEN');
 	}
 
-	if (($action == 'confirm_addconsumeline' && GETPOST('addconsumelinebutton') && $permissiontoadd)
-	|| ($action == 'confirm_addproduceline' && GETPOST('addproducelinebutton') && $permissiontoadd)) {
+	if ($action == 'confirm_addconsumeline' && GETPOST('addconsumelinebutton') && $permissiontoadd) {
 		$moline = new MoLine($db);
 
 		// Line to produce
 		$moline->fk_mo = $object->id;
 		$moline->qty = GETPOST('qtytoadd', 'int'); ;
 		$moline->fk_product = GETPOST('productidtoadd', 'int');
-		if (GETPOST('addconsumelinebutton')) {
-			$moline->role = 'toconsume';
-		} else {
-			$moline->role = 'toproduce';
-		}
+		$moline->role = 'toconsume';
 		$moline->origin_type = 'free'; // free consume line
```

Comment:
Can you fix this please

$line->qty-$alreadyconsumed

Thanks

Reply:
Excuse me, I'm quite new to git and php.
I don't really understand what you mean that I have to fix.
I thought the '$line->qty' is the total number of products required and the '$alreadyconsumed' the number of products you already used in a partial production. The difference between those 2 is what is still needed to produce the other products and finish the manufacturing order.
If you can explain a little more what the problem is, I will fix it.